diff --git a/build.xml b/build.xml
--- a/src/main/groovy/org/codehaus/gant/ant/Gant.java
+++ b/src/main/groovy/org/codehaus/gant/ant/Gant.java
@@ -202,7 +202,7 @@
    */
   //  Russel Winder rehacked the code provided by Eric Van Dewoestine.
   private void addAlmostAll ( final Project newProject , final Project oldProject ) {
-    @SuppressWarnings ( "unchecked" ) final Hashtable<String,String> properties = oldProject.getProperties ( ) ;
+    @SuppressWarnings ( "unchecked" ) final Hashtable<String,String> properties = (Hashtable) oldProject.getProperties ( ) ;
     final Enumeration<String> e = properties.keys ( ) ;
     while ( e.hasMoreElements ( ) ) {
       final String key = e.nextElement ( ) ;
