--- a/build.xml
+++ b/build.xmljavac2
@@ -1248,13 +1248,12 @@
                 <compilerarg value="kotlin/Metadata"/> <!-- Annotation to strip -->
                 <compilerarg value="kotlin/reflect/jvm/internal/impl/.*"/> <!-- Classes to strip from -->
                 <compilerarg value="${output}/kotlin-reflect-after-jarjar.jar"/>
-                <compilerarg value="${output}/kotlin-reflect-before-proguard.jar"/>
+                <compilerarg value="${kotlin-home}/lib/kotlin-reflect.jar"/>
                 <classpath>
                     <pathelement location="${idea.sdk}/lib/asm-all.jar"/>
                 </classpath>
             </kotlinc>

-            <shrink configuration="${basedir}/core/reflection.jvm/reflection.pro"/>
         </sequential>

         <sequential unless:true="${obfuscate.reflect}">
