--- a/platform/core-impl/src/com/intellij/openapi/progress/util/AbstractProgressIndicatorBase.java
+++ b/platform/core-impl/src/com/intellij/openapi/progress/util/AbstractProgressIndicatorBase.java
@@ -23,7 +23,6@@
 import com.intellij.openapi.progress.ProgressIndicator;
 import com.intellij.openapi.progress.ProgressManager;
 import com.intellij.openapi.util.UserDataHolderBase;
-import com.intellij.ui.mac.foundation.MacUtil;
 import com.intellij.util.containers.ContainerUtil;
 import com.intellij.util.containers.DoubleArrayList;
 import com.intellij.util.containers.Stack;
@@ -89,16 +89,10 @@
   }

   protected void startSystemActivity() {
-    myMacActivity = myShouldStartActivity ? MacUtil.wakeUpNeo(toString()) : null;
+    myMacActivity = null;
   }

   protected void stopSystemActivity() {
-    if (myMacActivity != null) {
-      synchronized (myMacActivity) {
-        MacUtil.matrixHasYou(myMacActivity);
-        myMacActivity = null;
-      }
-    }
   }

   @Override
